doc/Compilation.md

```diff
@@ -103,6 +104,28 @@ has three main ramifications:
    should ensure that all Tock developers are using the same version of the
    tools.
 
+### Special `.apps` section
+
+Tock kernels include a `.apps` section in the kernel .elf file that is at the
+same physical address where applications will be loaded. When compiling the
+kernel, this is just a placeholder and is not populated with any meaningful
+data. It exists to make it easy to update the kernel .elf file with an
+application binary to make a monolithic .elf file so that the kernel and apps
+can be flashed together.
+
+When the Tock build system creates the kernel binary, it explicitly removes this
+section so that the placeholder is not included in the kernel binary.
+
+To use the special `.apps` section, `objcopy` can replace the placeholder with an actual app binary. The general command looks like:
+
+```bash
+$ arm-none-eabi-objcopy --update-section .apps=libtock-c/examples/c_hello/build/cortex-m4/cortex-m4.tbf target/thumbv7em-none-eabi/release/stm32f412gdiscovery.elf target/thumbv7em-none-eabi/release/stm32f4discovery-app.elf
+```
+
+This replaces the placeholder section `.apps` with the "c_hello" application TBF
+in the stm32f412gdiscovery.elf kernel ELF, and creates a new .elf called
+`stm32f4discovery-app.elf`.
+
```

Comment:
Should there be a discussion here about what to do with that elf file in order to get it on a board? Maybe a comment that says "board-specific README files explain this process and how to load the ELF onto that particular board".

Reply:
What's missing right now is how I would load a monolithic kernel on something like Hail or an nRF dev kit where it isn't the default. Running `make flash` or `make program` after running this objcopy would just load a kernel with the apps section stripped. I would end up having to parse through the board make files for a bit to find some `objcopy -O binary` incantation followed by another JTAG incantation.

Reply:
Add a commit? I don't think we can promise that board readmes will explain this. My intent was to explain this enough that it is written down somewhere in some form, but not to imply this is the default or intended behavior.